vkcs/blockstorage/resource_volume.go

```diff
@@ -199,12 +209,31 @@ func resourceBlockStorageVolumeRead(ctx context.Context, d *schema.ResourceData,
 	d.Set("description", v.Description)
 	d.Set("snapshot_id", v.SnapshotID)
 	d.Set("source_vol_id", v.SourceVolID)
-	d.Set("metadata", v.Metadata)
 	d.Set("region", util.GetRegion(d, config))
+	d.Set("all_metadata", v.Metadata)
+
+	configMetadata := d.Get("metadata").(map[string]any)
+	intersectionMetadata := make(map[string]string, len(configMetadata))
+	for key := range configMetadata {
+		if _, exist := v.Metadata[key]; exist {
+			intersectionMetadata[key] = v.Metadata[key]
+		}
+	}
+	d.Set("metadata", intersectionMetadata)
 
 	return nil
 }
 
+type VolumeUpdateOpts struct {
```

Comment:
This should be placed into `vkcs/internal/services/blockstorage`.

Reply:
done